src/BloomExe/web/EnhancedImageServer.cs

```diff
@@ -140,8 +141,75 @@ public static SimulatedPageFile MakeSimulatedPageFileInBookFolder(HtmlDom dom, b
 			return new SimulatedPageFile() {Key = url};
 		}
 
+		/// <summary>
+		/// Producing PDF with gecko doesn't work on Windows for some reason when referencing
+		/// the localhost server.  So we need to use actual files, and refer to actual files in
+		/// the links to the css files.
+		/// </summary>
+		/// <remarks>
+		/// See http://jira.sil.org/browse/BL-932 for details on the bug.
+		/// </remarks>
+		public static SimulatedPageFile MakeRealPublishModeFileInBookFolder(HtmlDom dom)
+		{
+			var simulatedPageFileName = Path.ChangeExtension(Guid.NewGuid().ToString(), ".htm");
```

Comment:
John is keen that the temp file should NOT have an htm extension. In case Bloom crashes, a leftover temp file in the book folder is not a disaster...unless it has this extension, in which case, the user will no longer be able to open the book. I tried, and FF does not seem to care if you give it a url to a file with an extension like .tmp.